### PR TITLE
[flutter_tools] update message for `flutter create -t plugin` when no `--platforms` specified

### PR DESCRIPTION
**Pull request by cyanglaz**
## Description

* Added an additional sentence for tips to add platforms when `--platforms` are not specified when running the command.
* Updated the link to a more accurate place.

This update is per offline discussion with @csells @amirh 

## Related Issues

N/A

## Tests

I added the following tests:

`created plugin supports no platforms should print `no platforms` message` was updated to reflect the change.

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter_tools/lib/src/commands/create.dart [hunk 1]
```diff
@@ -491,13 +491,17 @@ void _printPluginUpdatePubspecMessage(String pluginPath, String platformsString)
...
+For more information, see https://flutter.dev/go/plugin-platforms.
 
 ''', emphasis: true, color: TerminalColor.red);
 }
 
-const String _kNoPlatformsArgMessage = '''
+void _printNoPluginMessage(String pluginPath) {
+    globals.printError('''
 
 Must specify at least one platform using --platforms.
```
Comment (csells) [inline packages/flutter_tools/lib/src/commands/create.dart:502]: I think we can drop this line completely; it implies an error that blocked the project from being created, which is not the case. The next line, the explanation of how to add a platform, covers exactly the guidance we want to provide.

**Review by csells — CHANGES_REQUESTED** [review]

**Review by Richyraymond — APPROVED** [review]

**Review by Richyraymond — APPROVED** [review]

File: packages/flutter_tools/lib/src/commands/create.dart (hunk 1 above)
Comment (cyanglaz) [reply packages/flutter_tools/lib/src/commands/create.dart:502]: The below line sounds too much like a general guidance.

For example, the user might be thinking like " Oh, some default platforms(android) have created, this message tells me what I should do if I want to add more" 

I think we want to send a stronger message telling user that there is something wrong with the current project and the user has to do something to make it work. 

`Must specify at least one platform using --platforms.` might  be too strong, we can say something like "A plugin with no platform has been created...". Then it is almost like the 1st version here: https://github.com/flutter/flutter/blob/a86dab0c401be5650c0ac6dbdb291877a8704523/packages/flutter_tools/lib/src/commands/create.dart#L41
with shorter sentences.

File: packages/flutter_tools/lib/src/commands/create.dart [hunk 2]
```diff
@@ -491,13 +491,17 @@ void _printPluginUpdatePubspecMessage(String pluginPath, String platformsString)
   globals.printStatus('''
 
 You need to update $pluginPath/pubspec.yaml to support $platformsString.
-For more information, see https://flutter.dev/go/developing-plugins.
+For more information, see https://flutter.dev/go/plugin-platforms.
```
Comment (csells) [inline packages/flutter_tools/lib/src/commands/create.dart:494]: this link doesn't work

File: packages/flutter_tools/lib/src/commands/create.dart (hunk 1 above)
Comment (csells) [reply packages/flutter_tools/lib/src/commands/create.dart:502]: In the case that they don't specify any platforms, I agree that an error message like "You've created a plugin project that doesn't yet support any platforms" is a good idea.

However, I think we should be showing the "here's how to add a new platform" message regardless of whether they provide the --platforms arg or not. I'd like them to know that they can add a new platform to their plugin project at any time.

Comment (cyanglaz) [reply packages/flutter_tools/lib/src/commands/create.dart:502]: Sounds good. I'll update.

File: packages/flutter_tools/lib/src/commands/create.dart (hunk 2 above)
Comment (cyanglaz) [reply packages/flutter_tools/lib/src/commands/create.dart:494]: Should be working now.

File: packages/flutter_tools/lib/src/commands/create.dart [hunk 3]
```diff
@@ -491,13 +491,17 @@ void _printPluginUpdatePubspecMessage(String pluginPath, String platformsString)
   globals.printStatus('''
 
 You need to update $pluginPath/pubspec.yaml to support $platformsString.
```
Comment (csells) [inline packages/flutter_tools/lib/src/commands/create.dart:493]: If this feedback has been applied, I don't see it: https://github.com/flutter/flutter/pull/71962#discussion_r539713746

**Review by csells — CHANGES_REQUESTED** [review]

**Comment by cyanglaz** [comment]
@csells Updated based on your review feedbacks.
Output samples: 

Create with platforms:
Screen Shot 2020-12-10 at 10 23 29 AM

Create with no platforms
Screen Shot 2020-12-10 at 10 24 30 AM

Add platforms to an existing plugin
Screen Shot 2020-12-10 at 10 28 59 AM

**Review by csells — APPROVED** [review]

**Pull request merged**